### PR TITLE
Fixed cardiac gamemode not working

### PR DESCRIPTION
**Pull request by jriyyya**
I completely forgot about the Cardiac gamemode! 😅
Please Review @chimosky

### CODE REVIEW

File: fallitem.py [hunk 1]
```diff
@@ -17,13 +17,18 @@
 import pygame
 
 
-class Spike:
+class FallItem:
 
-    def __init__(self, x, y, flip, speed, path_length):
+    def __init__(self, x, y, flip, speed, path_length, _type = "spike"):
```
Comment (chimosky) [inline fallitem.py:22]: `_type` is a kwarg so no need for spaces and why make it private?

File: fallitem.py [hunk 2]
```diff
@@ -17,13 +17,18 @@
...
         self.x = x
         self.initial = y
 
+        self.type = _type
```
Comment (chimosky) [inline fallitem.py:26]: This is what you should make private `self._type` as that's consistent with activities.

File: game.py [hunk 3]
```diff
@@ -26,7 +26,7 @@
 class Game:
 
     def __init__(self, bg_image_path, keymap, config,
-                 border_width=16, speed=7):
+                 border_width=16, speed=7, _type="spikes"):
```
Comment (chimosky) [inline game.py:29]: Here too.

File: game.py [hunk 4]
```diff
@@ -65,7 +65,13 @@ def __init__(self, bg_image_path, keymap, config,
...
 
         self.generator = Generator()
-        self.generator.configure(spikes_config)
+        if _type == "spikes":
+            self.generator.configure(spikes_config, "spike")
+        elif _type == "cardiac":
+            self.generator.configure(spikes_config, "heart")
+        else:
+            raise Exception("Incorrect game mode")
+
```
Comment (chimosky) [inline game.py:74]: Why this is pythonic, it's not consistent with Sugar activities. 
Instance variables are referred to by `self.name` as that's less ambiguous.

File: generator.py [hunk 5]
```diff
@@ -14,38 +14,51 @@
...
+        self.items_config = []
+        self.items = []
+        self.type = "spike"
 
-    def configure(self, config):
-        self.spikes = []
-        self.spikes_config = config
-        for _ in range(len(self.spikes_config)):
-            self.spikes.append([])
+    def configure(self, config, _type = "spike"):
```
Comment (chimosky) [inline generator.py:28]: Here too.

File: fallitem.py (hunk 2 above)
Comment (jriyyya) [reply fallitem.py:26]: I just checked where the underscore has to be for default keywords which cannot be used as variable name. Turned it the underscore should be present after the name, I will change it to `self.type_` then

File: fallitem.py (hunk 1 above)
Comment (jriyyya) [reply fallitem.py:22]: I just checked where the underscore has to be for default keywords which cannot be used as variable name. Turned it the underscore should be present after the name, I will change it to self.type_ then

File: game.py (hunk 4 above)
Comment (jriyyya) [reply game.py]: Hmm missed it, Corrected it.

File: fallitem.py (hunk 1 above)
Comment (chimosky) [reply fallitem.py]: Seeing as it's not a convention we use, don't use it.

Use the typical convention we use, which is to reassign it to a private instance variable.

**Comment by chimosky** [comment]
Tested eb1fde8, works as expected. Thanks!

**Pull request merged**